code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/EventListActivity.java

```diff
@@ -69,41 +91,177 @@ public void onClick(View v) {
+                            .collection(habitsColName)
+                            .document(habit_list.get(i))
+                            .collection(habitEventsColName);
+                    colRef.addSnapshotListener(new EventListener<QuerySnapshot>() {
+                        @Override
+                        public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable
+                                FirebaseFirestoreException error) {
+                            for (QueryDocumentSnapshot doc : queryDocumentSnapshots) {
+                                boolean duplicate_flag = false;
+                                for (int i = 0;i<eventDataList.size();i++) {
+                                    if(eventDataList.get(i).getEventId() == doc.getId()) {
+                                        duplicate_flag = true;
+                                        Log.d("same!", String.valueOf(doc.getId()));
+                                    }
+                                }
+                                if(duplicate_flag) {
+                                    continue;
+                                }
+                                Log.d(DB_TAG, String.valueOf(doc.getData().get("Habit")));
+                                String eventID = doc.getId();
+                                String habitID = (String) doc.getData().get("Habit");
+
+                                Log.d("eventID", String.valueOf(doc.getId()));
+                                Log.d("date", String.valueOf(doc.getData().get("startDate")));
+
+                                ArrayList<Integer> startDate = (ArrayList<Integer>) doc.getData().get("startDate");
+                                String comments = (String) doc.getData().get("comment");
+//                    String location = (String) doc.getData().get("location");
+//                    String image = (String) doc.getData().get("image");
+                                HabitEvent temp = new HabitEvent();
+                                temp.setStartDate(startDate);
+                                temp.setComment(comments);
+                                temp.setEventId(eventID);
+                                temp.setHabit(habitID);
+                                if(!(habitID=="")) {
+                                    eventDataList.add(temp);
+                                }
+
+                            }
+                            eventAdapter.notifyDataSetChanged(); // Notifying the adapter to render any new data fetched
+                            //from the cloud
+                        }
+                    });
+                }
+//                eventAdapter.notifyDataSetChanged(); // Notifying the adapter to render any new data fetched
+
+            }
+
+            @Override
+            public void onCallbackFailed() {
+
+            }
+        });
+//        String []habit_list = {"Habit 1","Habit 2","Habit 3"};
```

Comment:
What's all of this commented code for? Old testing maybe?

Reply:
Removed